utility_manager.py

```diff
@@ -0,0 +1,147 @@
+            back_button.id = f'{node_number}'
+        #if its form the add input page, return back there
+        else:
+            back_button = Button(text="Back", size_hint=(None,None), size=(40,20), on_press=self.plugin_player.node_widget_manager.add_input)
+            back_button.id = f'{node_number} {key_number}'
+        types_box.add_widget(back_button)
+        height += 20
+        
+        types_box.add_widget(Label(text="Add a new class type by filling in the blanks separated by a space", color=(0,0,0,1),
+                size_hint_y=None, height=20))
+        
+        height += 20
+
+        
+        #create the entry to add an extra input
+        new_type = BoxLayout(orientation='horizontal', size_hint_y=None, height=35, spacing = 0)
+        new_type_entry = TextInput(hint_text="from _____ import _____", multiline=False, size_hint_x=9/10)
+        new_type.add_widget(new_type_entry)
+
+        #set as global to pull from it when its submitted
+        self.custom_declaration_widget = new_type_entry
+
+        #add an import button
+        add_button = Button(text="Import",size_hint_x = 1/10, on_press=self.new_type)
+        #if its a property type assignt the id to route back to view_config
+        if(key_number == -1):
+            add_button.id = f'{node_number} {-1}'
+        #otherwise route to input page
+        else:
+            add_button.id = f'{node_number} {key_number}'
+        new_type.add_widget(add_button)
+
+        types_box.add_widget(new_type)
+        height += 35
+
+        types_box.add_widget(Label(text="Imported Types", font_size="20sp", size_hint_y=None, height=20))
+        height += 20
+
+        for imported_type in self.plugin_player.imported_classes:
+            types_box.add_widget(Label(text=imported_type, size_hint_y=None, height=15, color=(0,0,0,1)))
+            height += 15
+        
+        #add scrolling capabilities
+        if height < 450:
+            types_box.add_widget(Widget(size_hint_y=None, height=(450-height)))
+        scroll_view = ScrollView(scroll_y=0, do_scroll_y=True, scroll_type=['content'])
+        scroll_view.add_widget(types_box)
+
+        self.plugin_player.create_popup(scroll_view, "Class Types", False)
+
+    #define a new class type for custom inputs and property types
+    def new_type(self, instance):
+        try:
+            import_name = self.custom_declaration_widget.text.split()[0]
+            class_name = self.custom_declaration_widget.text.split()[1]
+            if class_name in self.plugin_player.imported_classes:
+                self.plugin_player.addMessage(f"Class type {class_name} previously imported")
+                return
+        except Exception as e:
+            self.plugin_player.addMessage("Invalid Entry, enter the blanks with a space in between from the Python import statement")
+            self.plugin_player.addMessage(e)
+            return
```

Comment:
```suggestion
        try:
            import_name = self.custom_declaration_widget.text.split()[0]
            class_name = self.custom_declaration_widget.text.split()[1]
        except Exception as e:
            self.plugin_player.addMessage("Invalid Entry, enter the blanks with a space in between from the Python import statement")
            self.plugin_player.addMessage(e)
            return
        if class_name in self.plugin_player.imported_classes:
            self.plugin_player.addMessage(f"Class type {class_name} previously imported")
            return
```
The `if` doesn't need to be nested into the `try` block.